lib/Loans/OpenLoans/OpenLoans.js

```diff
@@ -288,6 +269,23 @@ class OpenLoans extends React.Component {
     this.props.onClickViewLoanActionsHistory(e, loan);
   }
 
+  getOpenRequestsCount() {
+    const q = this.state.loans.map(loan => {
+      return `itemId==${loan.itemId}`;
+    }).join(' or ');
+
+    const query = `(${q}) and status==("Open - Awaiting pickup" or "Open - Not yet filled") sortby requestDate desc`;
+    this.props.mutator.requests.reset();
+    this.props.mutator.requests.GET({ params: { query } }).then((requestRecords) => {
+      let requestCount = 0;
+      const requestCountObject = requestRecords.reduce((map, record) => {
+        map[record.itemId] = map[record.itemId] ? map[record.itemId]++ : ++requestCount;
```

Comment:
@aditya-matukumalli I'm probably missing something here but what is the purpose of `requestCount`? Why would we want to assign `requestCount` to  `map[record.itemId]`?

Reply:
@mkuklis requestCount here represents the number of requests open against individual item. I am storing a map of the itemId and the corresponding number of requests open against them. I guess I could just assign 1 instead of `++requestCount` as  `map[record.itemId]++` should take care of the count after the initial increment. That way I could just get rid of that variable. Does this address your concern?

Reply:
@aditya-matukumalli yes replacing it with just `1` and getting rid of that var was what I was thinking.

Reply:
@mkuklis yeah, added that in. Working on @doytch's change requests